Impossible to deploy a token

### Describe the bug

Trying to deploy a token like 
```ts
class ZkNoidCoin extends TokenContract implements Erc20Like {
  // constant supply
  SUPPLY = UInt64.from(1_000_000_000n * 10n ** 9n);
``` 
where `TokenContract` extends SmartContract I got error
```
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/andrey/projects/zknoid-token-2/contracts/build/src/undefined' imported from /Users/andrey/.nvm/versions/node/v18.19.1/lib/node_modules/zkapp-cli/src/lib/deploy.js
```

The error is in function `findSmartContracts` [main/src/lib/deploy.js#L677](https://github.com/o1-labs/zkapp-cli/blob/main/src/lib/deploy.js#L677|main/src/lib/deploy.js#L677) that expects the following pattern:
```
export async function findSmartContracts(path) {
   ...
    // 
    let results = str.matchAll(/class (\w*) extends SmartContract/gi);
}

```

If contract extends not the SmartContract directly but any contract that extends SmartContract (e.g. TokenContract from o1js), it's not found by the deployer

### Steps to Reproduce

Create contract like here https://github.com/o1-labs/o1js/blob/main/src/examples/zkapps/dex/erc20.ts
Try to deploy it

### System Info

```shell
System:
    OS: macOS 14.4
    CPU: (16) arm64 Apple M3 Max
  Binaries:
    Node: 18.19.1 - ~/.nvm/versions/node/v18.19.1/bin/node
    Yarn: 1.22.21 - ~/.nvm/versions/node/v18.19.1/bin/yarn
    npm: 10.2.4 - ~/.nvm/versions/node/v18.19.1/bin/npm
  npmPackages:
    o1js: 1.1.0
  npmGlobalPackages:
    zkapp-cli: 0.20.1
```


### Severity

serious, but I can work around it

### Additional Information

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -17,6 +17,12 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 ## Unreleased
 
+## [0.20.2](https://github.com/o1-labs/zkapp-cli/compare/0.20.1...0.20.2) - 2024-05-07
+
+### Fixed
+
+- [Hotfix] SmartContract classes lookup for deployment. [#637](https://github.com/o1-labs/zkapp-cli/pull/637)
+
 ## [0.20.1](https://github.com/o1-labs/zkapp-cli/compare/0.20.0...0.20.1) - 2024-04-30
 
 ### Added

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "zkapp-cli",
-  "version": "0.20.1",
+  "version": "0.20.2",
   "description": "CLI to create zkApps (zero-knowledge apps) for Mina Protocol",
   "homepage": "https://github.com/o1-labs/zkapp-cli/",
   "keywords": [

--- a/src/lib/deploy.js
+++ b/src/lib/deploy.js
@@ -674,9 +674,11 @@ export async function findSmartContracts(path) {
 
   for (const file of files) {
     const str = fs.readFileSync(file, 'utf-8');
-    let results = str.matchAll(/class (\w*) extends SmartContract/gi);
+    // TODO: Implement better SmartContract classes lookup.
+    // https://github.com/o1-labs/zkapp-cli/issues/636
+    let results = str.matchAll(/class (\w+) (extends|implements) (\w+)/gi);
     results = Array.from(results) ?? []; // prevent error if no results
-    results = results.map((result) => result[1]); // only keep capture groups
+    results = results.map((result) => result[1]); // only keep first capture group, the class name
     smartContracts.push(...results);
   }
   return smartContracts;

